server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -195,7 +253,31 @@ public RecoveryStats getRecoveryStats() {
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         stats.writeTo(out);
+
+        if (out.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
As explained in StreamInput, this is the new behaviour introduced as an optimisation.